devicetypes/timevalve-gaslock-t-08/timevalve-smart.src/timevalve-smart.groovy

```diff
@@ -0,0 +1,281 @@
+    } else {
+        map.value = cmd.batteryLevel
+    }
+    
+    log.debug "battery - ${map.value}${map.unit}"
+    // Store time of last battery update so we don't ask every wakeup, see WakeUpNotification handler
+    state.lastbatt = new Date().time
+    createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+  //log.debug "zwaveEvent - ${device.displayName}: ${cmd}"
+  createEvent(descriptionText: "${device.displayName}: ${cmd}")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv1.ConfigurationReport cmd) {
+	def result = []
+	log.info "zwave.configurationV1.configurationGet - " + cmd
+    def array = cmd.configurationValue
+    def value = ( (array[0] * 0x1000000) + (array[1] * 0x10000) + (array[2] * 0x100) + array[3] ).intdiv(60)
+    if (device.currentValue("switch") == "on") {
+    	value = ( (array[0] * 0x1000000) + (array[1] * 0x10000) + (array[2] * 0x100) + array[3] ).intdiv(60)
+    } else {
+    	value = 0
+    }
+    
+    if (device.currentValue('contact') == 'open') {
+    	def hour = value.intdiv(60);
+        def min = (value % 60).toString().padLeft(2, '0');
+        def text = "${hour}:${min}M"
+        
+        log.info "remain - " + text
+        result.add( createEvent(name: "remaining", value: value, displayed: false, isStateChange: true) )
+        result.add( createEvent(name: "remainingText", value: text, displayed: false, isStateChange: true) )
+    } else {
+    	result.add( createEvent(name: "timeout", value: value, displayed: false, isStateChange: true) )
+    }
+    
+    /*
+    if (cmd.parameterNumber == 0x01) { // timeout
+    	result.add( createEvent(name: "timeout", value: value, displayed: false, isStateChange: true) )
+        
+        
+        // timeout 설정 시에도 
+        def hour = value.intdiv(60);
+        def min = (value % 60).toString().padLeft(2, '0');
+        def text = "${hour}:${min}M"
+        
+        result.add( createEvent(name: "remaining", value: value, displayed: false, isStateChange: true) )
+        result.add( createEvent(name: "remainingText", value: text, displayed: false, isStateChange: true) )
+    } else if (cmd.parameterNumber == 0x03) { // remaining
+        def hour = value.intdiv(60);
+        def min = (value % 60).toString().padLeft(2, '0');
+        def text = "${hour}:${min}M"
+        
+        debug.info "remain - " + text
+        result.add( createEvent(name: "remaining", value: value, displayed: false, isStateChange: true) )
+        result.add( createEvent(name: "remainingText", value: text, displayed: false, isStateChange: true) )
+    }
+    */
+    return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+	def type = cmd.notificationType
+    if (type == cmd.NOTIFICATION_TYPE_HEAT) {
+    	log.info "NotificationReport - ${type}"
+        createEvent(name: "temperature", value: 80, descriptionText: "${device.displayName} is over heat!", displayed: true, isStateChange: true)
```

Comment:
It might be more appropriate to set a larger value (like 999), and then define the Units for this. Such as:

createEvent(name: "temperature", value: 999, unit: "C", descriptionText: "${device.displayName} is over heat!", displayed: true, isStateChange: true)